test/socket.push-pull.js

```diff
@@ -38,4 +38,89 @@ describe('socket.push-pull', function(){
     });
   });
 
-});
\ No newline at end of file
+
+  it('should not emit messages after pasue()', function(done){
+      var push = zmq.socket('push')
+        , pull = zmq.socket('pull');
+
+      var n = 0;
+
+      pull.on('message', function(msg){
+        if(n++ === 0) {
+          msg.toString().should.equal('foo');
+        }
+        else{
+          should.not.exist(msg);
+        }
+      });
+
+      var addr = "inproc://pause_stuff";
+
+      pull.bind(addr, function(){
+        push.connect(addr);
+
+        push.send('foo');
+        pull.pause()
+        push.send('bar');
+        push.send('baz');
+      });
+
+      setTimeout(function (){
+        pull.close();
+        push.close();
+        done();
+      }, 100);
+  });
+
+
+
+  it('should emit messages after resume()', function(done){
+    var push = zmq.socket('push')
+      , pull = zmq.socket('pull');
+
+    var n = 0;
+
+    checkNoMessages = function (msg){
+      should.not.exist(msg);
+    }
+
+    checkMessages = function (msg){
```

Comment:
Dito, this is an anonymous function without the need for one (makes for horrible stack traces).